modules/ingestor/src/main/scala/mongo.ublog.scala

```diff
@@ -0,0 +1,122 @@
+package lila.search
+package ingestor
+
+import cats.effect.IO
+import cats.syntax.all.*
+import com.mongodb.client.model.changestream.FullDocument
+import com.mongodb.client.model.changestream.OperationType.*
+import mongo4cats.bson.Document
+import mongo4cats.database.MongoDatabase
+import mongo4cats.models.collection.ChangeStreamDocument
+import mongo4cats.operations.{ Aggregate, Filter, Projection }
+import org.typelevel.log4cats.syntax.*
+import org.typelevel.log4cats.{ Logger, LoggerFactory }
+
+import java.time.Instant
+import scala.concurrent.duration.*
+
+import Repo.{ *, given }
+
+object UblogRepo:
+
+  private val interestedOperations = List(DELETE, INSERT, REPLACE, UPDATE).map(_.getValue)
+
+  private val interestedFields =
+    List(_id, F.markdown, F.title, F.intro, F.topics, F.blog, F.live, F.livedAt, F.language, F.quality)
+  private val postProjection = Projection.include(interestedFields)
+
+  private val interestedEventFields =
+    List("operationType", "clusterTime", "documentKey._id") ++ interestedFields.map("fullDocument." + _)
+  private val eventProjection = Projection.include(interestedEventFields)
+
+  private def aggregate() =
+    Aggregate
+      .matchBy(Filter.in("operationType", interestedOperations))
+      .combinedWith(Aggregate.project(eventProjection))
+
+  def apply(mongo: MongoDatabase[IO], config: IngestorConfig.Ublog)(using
+      LoggerFactory[IO]
+  ): IO[Repo[UblogSource]] =
+    given Logger[IO] = LoggerFactory[IO].getLogger
+    // (mongo.getCollection("ublog_blog"), mongo.getCollection("ublog_post")).mapN(apply(config))
+    mongo.getCollection("ublog_post").map(apply(config))
+
+  def apply(config: IngestorConfig.Ublog)(
+      // blogs: MongoCollection,
+      posts: MongoCollection
+  )(using Logger[IO]): Repo[UblogSource] = new:
+
+    def fetch(since: Instant, until: Instant) =
+      val filter = range(F.livedAt)(since, until.some)
+      fs2.Stream.eval(info"Fetching blog posts from $since to $until") *>
+        posts
+          .find(filter)
+          .projection(postProjection)
+          .boundedStream(config.batchSize)
+          .filter(_.isLive)
```

Comment:
We use `isLive` to partition `toDelete` and `toIndex` few lines below. So I guess this filter is an oversight?

Reply:
yus